src/Sarif/Visitors/InsertOptionalDataVisitor.cs

```diff
@@ -39,24 +50,9 @@ public override Run VisitRun(Run node)
                 visitor.VisitRun(node);
             }
 
-            if (node.Files != null)
-            {
-                // Note, we modify this collection as  we enumerate it.
-                // Hence the need to convert to an array here. Otherwise,
-                // the standard collection enumerator will throw an
-                // exception after we touch the collection.
-                var keys = node.Files.Keys.ToArray();
-                foreach (string key in keys)
-                {
-                    var value = node.Files[key];
-                    if (value != null)
-                    {
-                        node.Files[key] = VisitDictionaryValueNullChecked(key, value);
-                    }
-                }
-            }
+            Run visited = base.VisitRun(node);
```

Comment:
>visited [](start = 16, length = 7)

Ultra-nit (I'm not even asking you to change this, since you're clearly expressing a personal preference here): The method declaration already tells us it returns a `Run`, so IMO we don't need the extra documentation we get by assigning the result of this call to a temp variable.

Again, that's just me: I leave the line `return base.VisitRun(node)` alone.